### ISSUE TITLE
[Bug]: Unexpected server errors

### ISSUE THREAD

**Issue by apepkuss** [milvus-io/milvus-sdk-rust#48]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I tried to put a line of code (for checking the response status) in the [Collection::insert](https://github.com/milvus-io/milvus-sdk-rust/blob/main/src/collection.rs#L362-L397) method before returning from the method:

```rust
pub async fn insert(
    &self,
    fields_data: Vec<FieldColumn>,
    partition_name: Option<&str>,
) -> Result<crate::proto::milvus::MutationResult> {

    ...

    status_to_result(&result.status)?;  // check the response status

    Ok(result)
}
```

After that, I run the following test code derived from [this test function](https://github.com/milvus-io/milvus-sdk-rust/blob/main/tests/collection.rs#L33). 

```rust
#[tokio::test]
async fn collection_basic() -> Result<()> {
    let collection = create_test_collection().await?;

    let embed_data = gen_random_f32_vector(5);

    let embed_column = FieldColumn::new(
        collection.schema().get_field(DEFAULT_VEC_FIELD).unwrap(),
        embed_data,
    );

    collection.insert(vec![embed_column], None).await?;

    clean_test_collection(collection).await?;
    Ok(())
}
```

** Bugs **

- Bug-1: The following server error was returned:
   ```bash
   Error: Server(UnexpectedError, "num_rows(0) should be greater than 0")
   ```

   The root cause of the issue is the [FieldColumn::len](https://github.com/milvus-io/milvus-sdk-rust/blob/main/src/data.rs#L142-L144) method. The correct implementation should be 

   ```rust
    #[inline]
    pub fn len(&self) -> usize {
        let c = self.value.len() / self.dim as usize;
        if self.value.len() % self.dim as usize == 0 {
            c
        } else {
            c + 1
        }
    }
   ```
- Bug-2: After fixing the first bug, still run the test code above, another server error was raised:
   ```bash
   Error: Server(UnexpectedError, "the length(5) of float data should divide the dim(128)")
   ```
   Could you please help check the issue? The original [Collection::insert](https://github.com/milvus-io/milvus-sdk-rust/blob/main/src/collection.rs#L362-L397) method does not contain the response status checking code. Is it correct or not? Looking forward to your reply and thanks a lot!

### Environment details

```markdown
- Hardware/Software conditions (OS, CPU, GPU, Memory):
  Ubuntu-22.04

- Method of installation:
  
  Follow the [steps](https://milvus.io/docs/install_standalone-docker.md) in the official webpage
```

**Comment by yah01** [comment]
The `create_test_collection()` creates a collection with schema:
https://github.com/milvus-io/milvus-sdk-rust/blob/d9af92c0bbccc39c391861dca42de99a59c24119/tests/common/mod.rs#L22-L29
we divides the f32 array to `len() / dim` vectors, each one with length of `dim`. In your code, the dim is 128, but you are tring to insert only 5 f32 values. 

`gen_random_f32_vector(len)` generates `len` f32 values, not `len` vectors

**Comment by yah01** [comment]
You could ref the example in test:
https://github.com/milvus-io/milvus-sdk-rust/blob/d9af92c0bbccc39c391861dca42de99a59c24119/tests/collection.rs#L36-L43
